Clarifying when to use `--warn-error-options` (#7168)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/warnings.md

sidebar: "Warnings"
---

Enabling `WARN_ERROR`config or setting the `--warn-error` flag will convert dbt warnings into errors. Any time dbt would normally warn, it will instead raise an error. Examples include `--select` criteria that selects no resources, deprecations, configurations with no associated models, invalid test configurations, or tests and freshness checks that are configured to return warnings.

>  You would commonly use the `--warn-error` flag to promote test warnings from `severity: warn` to errors, but it actually affects all warning types, including:

Reviewer: would we normally just use this for test warnings? for some reason, i thought it could also be used in general for all warnings since its a flag you can pass as a command

>  You would commonly use the `--warn-error` flag to promote test warnings from `severity: warn` to errors, but it actually affects all warning types, including:

Reviewer: would we normally just use this for test warnings? for some reason, i thought it could also be used in general for all warnings since its a flag you can pass as a command. If so, i'm not sure we should say that's its used generally to promote test warnings to errors 

```suggestion
You can use the `--warn-error` flag to promote all warnings to errors, such as:
```

* Test warnings (for example, `LogTestResults`)
* Jinja-level warnings (for example, `exceptions.warn`, or `JinjaLogWarning`)
* Selection issues (for example, `NoNodesForSelectionCriteria`)
* Adapter deprecation warnings (for example, `AdapterDeprecationWarning`)

> Consider using [`--warn-error-options`](#using-warn_error_options-for-targeted-warnings) for more targeted control over which warnings are treated as errors.

Reviewer: ```suggestion
Consider using [`--warn-error-options`](#using---warn-error-options-for-targeted-warnings) for more targeted control over which warnings are treated as errors. As an example, if you'd like to only promote test warnings to errors, set that flag to `--warn-error-options '{"error": ["LogTestResult"]}'`.
```
